sql/connect/common/pom.xml

```diff
@@ -142,8 +218,26 @@
                         <includes>
                             <include>org.spark-project.spark:unused</include>
                             <include>org.apache.tomcat:annotations-api</include>
+                            <include>com.google.guava:*</include>
                         </includes>
                     </artifactSet>
+                    <filters>
+                        <filter>
+                            <artifact>com.google.guava:guava</artifact>
+                            <excludes>
+                                <exclude>com/google/thirdparty/**</exclude>
+                            </excludes>
+                        </filter>
+                    </filters>
+                    <relocations>
+                        <relocation>
+                            <pattern>com.google.common</pattern>
+                            <shadedPattern>${spark.shade.packageName}.connect.guava</shadedPattern>
+                            <includes>
+                                <include>com.google.common.**</include>
+                            </includes>
+                      </relocation>
+                    </relocations>
```

Comment:
I'm trying to understand the changes here and in the client pom, which address the `org/spark project/guava/cache/CacheLoader` error. Could you help explain your thought process?

Reply:
Note that the error refers to missing `org/sparkproject/guava/cache/CacheLoader` and not to `org/sparkproject/connect/guava/cache/CacheLoader` meaning that  the relocation still happens for guava classes in common jar as declared in parent pom. The change in the PR overwrites relocation rule defined in the parent  Spark pom with the one that was declared in the client pom. It solves the error as now relocated classes are part of the common jar.

Reply:
Ah... ouch. Great catch and nice fix!

Reply:
Can you explain this in some documentation? It is very easy for others to get tripped up by this.

Reply:
Do you refer to Spark Connect README or something else? Please suggest where that should be documented.

As PR is already merged, can you open JIRA.

Reply:
You can use this JIRA for this type of follow-ups.

Reply:
Please suggest what needs to be documented and where.

Reply:
IMO, it will be better to use JIRA to discuss ^^^ instead of the closed PR, though I am open to do this on the PR as well.

Reply:
> Please suggest what needs to be documented and where.

@hvanhovell